Yoast/wordpress-seo
# Implement internationalization for yoast-components

## Summary

This PR can be summarized in the following changelog entry:

* _not applicable_

## Relevant technical choices:

* Parse the translations in `yoast-components` using the babel plugin
`@wordpress/babel-plugin-makepot`. This plugin will parse `__`, `_x`,
`_n` and `_nx` in our JavaScript files. The script outputs to
`gettext.pot`, there was no clean way to change this to a file inside
`/languages`.
* Introduce a new class WPSEO_Admin_Asset_Yoast_Components_L10n
responsible for localizing the translations of yoast-components. It
can localize it for any script handle.
* Drop the conversion from .po to .json files and download the .json
files from GlotPress instead.
* Because yoast-components still has other ways to manage the
translations we need to combine two .pot file into one.
* Change `i18n-clean-json` to allow changing the text domain.

## Test instructions

This PR can be tested by following these steps:

* Make sure all our JavaScript still works

* When you link `yoast-components` to the latest develop no errors should show in the console about the `yoast-components` translations missing in our JavaScript.

If you encounter any errors make sure to first clean your `languages/` folder to make sure the tools run correctly.

## Quality assurance
